near-sdk/tests/store_performance_tests.rs

```diff
@@ -0,0 +1,268 @@
+        assert!(
+            total_gas > NearGas::from_tgas(90).as_gas(),
+            "not enough gas consumed: {}, adjust the number of iterations to spot regressions",
+            NearGas::from_gas(total_gas)
+        );
+    }
+
+    // contains
+    for col in Collection::iter().filter(|col| {
+        // No `contains` in vector.
+        !matches!(col, Collection::Vector)
+    }) {
+        let account_pool = account_pool.clone();
+        let contract_id = contract_id.clone();
+        let mut total_gas: u64 = 0;
+        let mut futures = JoinSet::new();
+
+        for val in 0..17 {
+            let account: Account = account_pool[val].clone();
+            let txn = account
+                .call(&contract_id.clone(), "exec")
+                .args_json((col, Op::Contains(3)))
+                .max_gas()
+                .transact();
+            futures.spawn(txn);
+        }
+        while let Some(res) = futures.join_next().await {
+            total_gas += res??.unwrap().total_gas_burnt.as_gas();
+        }
+
+        assert!(
+            total_gas < NearGas::from_tgas(100).as_gas(),
+            "performance regression: {}",
+            NearGas::from_gas(total_gas)
+        );
+        assert!(
+            total_gas > NearGas::from_tgas(90).as_gas(),
+            "not enough gas consumed: {}, adjust the number of iterations to spot regressions",
+            NearGas::from_gas(total_gas)
+        );
+    }
+
+    // flush
+    for col in Collection::iter().filter(|col| {
+        // LookupSet is not flushable.
+        !matches!(col, Collection::LookupSet)
+    }) {
+        let account_pool = account_pool.clone();
+        let contract_id = contract_id.clone();
+        let mut total_gas: u64 = 0;
+        let mut futures = JoinSet::new();
+
+        for val in 0..17 {
+            let account: Account = account_pool[val].clone();
+            let txn = account
+                .call(&contract_id.clone(), "exec")
+                .args_json((col, Op::Flush))
+                .max_gas()
+                .transact();
+            futures.spawn(txn);
+        }
+        while let Some(res) = futures.join_next().await {
+            total_gas += res??.unwrap().total_gas_burnt.as_gas();
+        }
+
+        assert!(
+            total_gas < NearGas::from_tgas(100).as_gas(),
+            "performance regression: {}",
+            NearGas::from_gas(total_gas)
+        );
+        assert!(
+            total_gas > NearGas::from_tgas(90).as_gas(),
+            "not enough gas consumed: {}, adjust the number of iterations to spot regressions",
+            NearGas::from_gas(total_gas)
+        );
+    }
+
+    // remove
+    for col in Collection::iter() {
+        let account_pool = account_pool.clone();
+        let contract_id = contract_id.clone();
+        let mut total_gas: u64 = 0;
+        let mut futures = JoinSet::new();
+
+        // Can't use more than 15, because that's how much was inserted.
+        for val in 0..15 {
```

Comment:
This could be a const that is shared between insert code and this one rather than a comment